### PR TITLE
Filter by domain 33314

### PR DESCRIPTION
**Pull request by joshnezianya**
## Description
Created feature to toggle filter menu on selection of domain.

## Motivation and Context
https://cm-jira.usa.gov/browse/IAE-33314

## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [X ] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Screenshots (if appropriate):
image
image

## Which browsers have you tested?
- [ X] Internet Explorer 11
- [ X] Edge
- [X ] Chrome
- [x] Firefox
- [x] Safari

## Checklist:

- [ ] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ ] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ X] All new and existing tests passed.
- [X ] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts [hunk 1]
```diff
@@ -183,9 +192,9 @@ export class SideNavigationSampleComponent implements AfterViewInit, OnInit {
   ];
 
 
-  public pageHeader: string;
-  public linkEvent = new BehaviorSubject<object>(null);
-  sideNavExpanded = true;
+  // public pageHeader: string;
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts:195]: please remove unused code

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html [hunk 2]
```diff
@@ -2,17 +2,20 @@
...
                     </sds-accordion-item-header>
-                    <sds-side-navigation #sideNav (linkEvent)="linkEvent.next($event)" [model]="model">
+                    <sds-side-navigation #sideNav  [model]="model">
                     </sds-side-navigation>
                 </sds-accordion-item>
-                <sds-accordion-item>
+                <sds-accordion-item #filtersAccordion>
                     <sds-accordion-item-header> Filters </sds-accordion-item-header>
-                    <sds-filters [form]="form" [fields]="fields" [model]="model1" (filterChange)="filterChange$.next($event)"></sds-filters>
+                    <sds-filters [form]="form" [fields]="fields" [model]="model1" (filterChange)="filterChange$.next($event)">
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html:15]: please use more descriptive model names

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html [hunk 3]
```diff
@@ -2,17 +2,20 @@
     <sds-page-options class="sam-page__options">
         <sds-toolbar [expanded]="sideNavExpanded" #toolbar="sdsToolbar">
             <sds-accordion multi="true">
-                <sds-accordion-item [expanded]="sideNavExpanded">
+                <sds-accordion-item #navigationAccordion [expanded]="navigationExpanded">
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html:5]: Styling is off in Chrome -- possibly incorrect class names
Screen Shot 2020-06-16 at 11 10 07 AM

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html (hunk 2 above)
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html:15]: filterChange is unnecessary/doesn't do anything

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html [hunk 4]
```diff
@@ -2,17 +2,20 @@
...
+                    <sds-side-navigation #sideNav  [model]="model">
                     </sds-side-navigation>
                 </sds-accordion-item>
-                <sds-accordion-item>
+                <sds-accordion-item #filtersAccordion>
                     <sds-accordion-item-header> Filters </sds-accordion-item-header>
-                    <sds-filters [form]="form" [fields]="fields" [model]="model1" (filterChange)="filterChange$.next($event)"></sds-filters>
+                    <sds-filters [form]="form" [fields]="fields" [model]="model1" (filterChange)="filterChange$.next($event)">
+                    </sds-filters>
+                    <div class="grid-row margin-top-1">
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html:17]: Remove - divs shouldn't be used just for styling

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { SideNavigationModel, NavigationMode, NavigationLink } from '@gsa-sam/co
 import { ActivatedRoute } from '@angular/router';
 import { FormlyFieldConfig, FormlyFormOptions } from '@ngx-formly/core';
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts:5]: remove unused dependency

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts [hunk 6]
```diff
@@ -289,19 +299,28 @@ export class SideNavigationSampleComponent implements AfterViewInit, OnInit {
     this.change.detectChanges();
 
     this.activeRoute.queryParams.subscribe(queryParams => {
+      console.log(queryParams)
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts:302]: remove all instances of console.log

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts [hunk 7]
```diff
@@ -183,9 +192,9 @@ export class SideNavigationSampleComponent implements AfterViewInit, OnInit {
   ];
 
 
-  public pageHeader: string;
-  public linkEvent = new BehaviorSubject<object>(null);
-  sideNavExpanded = true;
+  // public pageHeader: string;
+  // public linkEvent = new BehaviorSubject<object>(null);
+  // sideNavExpanded = true;
   ngOnInit() {
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts:198]: delete -- ngOnInit is unused without unnecessary filterChange$

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html (hunk 3 above)
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html:5]: Filter isn't changing heading navigation title 
Screen Shot 2020-06-16 at 11 22 59 AM

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts [hunk 8]
```diff
@@ -289,19 +299,28 @@ export class SideNavigationSampleComponent implements AfterViewInit, OnInit {
...
           this.findItemByQueryString(this.model.navigationLinks);
           this.sideNav.select(this.selectedId);
+          this.filtersAccordion.toggle();
         }
       } else {
-        this.pageHeader = 'Unknown';
+        this.pageHeader = 'Select by Domain';
+      }
+      console.log(this.navigationAccordion.expanded)
+      if(!this.filtersAccordion.expanded) {
```
Comment (ascreven-gsa) [inline libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts:314]: Refactor for speed and readability by removing from ngAfterViewInit so it isn't called after any change

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts (hunk 1 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts]: This has been removed.

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html (hunk 2 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html]: I actually didn't set these models up. I can make the change though. Is there a reference to naming conventions that you suggest I follow?

Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html]: I can remove this line and instances of filter change, but this isn't code that I wrote.

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html (hunk 4 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html]: This has been removed, but I don't believe that I wrote this code either.

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts (hunk 5 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts]: Removed.

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts (hunk 6 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts]: Removed.

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts (hunk 7 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts]: Removed.

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html (hunk 3 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.html:5]: I've made the update and it should work now.

File: libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts (hunk 8 above)
Comment (joshnezianya) [reply libs/component-sample/src/lib/feature/side-navigation-sample/side-navigation-sample.component.ts]: I've refactored the code here, but I didn't remove it from ngAfterViewInit. I found a different solutions. Let me know if the alternative is okay.

File: libs/documentation/src/lib/components/sidenavigation/sidenavigation.module.ts [hunk 9]
```diff
@@ -16,15 +16,8 @@ import { SideNavigationOptional } from './demos/optional/sidenavigation-optional
 
 declare var require: any;
 const DEMOS = {
```
Comment (davereed) [inline libs/documentation/src/lib/components/sidenavigation/sidenavigation.module.ts:18]: If you are removing basic, you should remove the files and folder as well.

File: libs/packages/components/src/lib/side-navigation/side-navigation.component.ts [hunk 10]
```diff
@@ -141,20 +154,19 @@ export class SdsSideNavigationComponent {
...
    * Link clicked and emits the link data into an event
-   * @param link 
+   * @param link
    */
-  linkClickEvent(link: INavigationLink) {
-    this.linkEvent.emit(link);
+  navigation(link: INavigationLink) {
+    this.navigate.emit(link);
     return false;
   }
```
Comment (davereed) [inline libs/packages/components/src/lib/side-navigation/side-navigation.component.ts:165]: Revert this change back to being named linkEvent, this will prevent a breaking change.

File: libs/packages/components/src/lib/side-navigation/side-navigation.component.ts [hunk 11]
```diff
@@ -168,17 +180,20 @@ export class SdsSideNavigationComponent {
         url += '&' + queryParams;
       }
     }
+    console.log(url);
```
[the remark this answers is not in the document]
Comment (davereed) [reply libs/packages/components/src/lib/side-navigation/side-navigation.component.ts]: Any update on this?

[the remark this answers is not in the document]
Comment (davereed) [reply libs/packages/components/src/lib/side-navigation/side-navigation.component.ts]: Any update on this?
